src/test/groovy/nu/studer/gradle/jooq/JooqFuncTest.groovy

```diff
@@ -54,6 +54,20 @@ tasks.named('generateJooq').configure { outputs.cacheIf { true } }
         result.task(':generateJooq').outcome == TaskOutcome.SUCCESS
     }
 
+    void "can invoke jOOQ task from configuration DSL with configuration name omitted from task name for 'main' configuration and jOOQ v3.16+"() {
+        given:
+        buildFile << buildWithJooqPluginDSL(null, null, null, '3.16.3', null)
+
+        when:
+        def result = runWithArguments('generateJooq')
+
+        then:
+        fileExists('build/generated-src/jooq/main/nu/studer/sample/jooq_test/tables/Foo.java')
+
+        and:
+        result.task(':generateJooq').outcome == TaskOutcome.SUCCESS
+    }
+
```

Comment:
Big thanks for adding a test for validating this! :+1: :tada: 

I can't get this to fail though. :thinking: Here's what I tried: I checked out the branch with my #206 changes. Then I added this test and executed it like this: `./gradlew test --info --tests "*3.16*"`. It was successful, even though the `jakarta.xml` changes weren't present.

By changing the `api` dependency for jOOQ, I was able to trigger the test failure though, i.e. this diff:

```diff
diff --git build.gradle build.gradle
index c257933..df59b75 100644
--- build.gradle
+++ build.gradle
@@ -13,7 +13,7 @@ group = 'nu.studer'
 version = '6.0.2-DEV'
 
 dependencies {
-    api 'org.jooq:jooq-codegen:3.15.1'
+    api 'org.jooq:jooq-codegen:3.16.3'
 
     implementation 'javax.xml.bind:jaxb-api:2.3.1'
     implementation 'org.glassfish.jaxb:jaxb-core:2.3.0.1'
```

Reply:
You are right, there is something strage here. It seems when executing the tests the jOOQ version in the plugin's classpath stays on 3.15.1. This 3.15.1 version is used to generate config.xml then after the plugin invokes `GenerationTool` with a different jOOQ version. It looks like a normal behavior for the plugin.

The bug happens when I apply the plugin on my project. It's a kotlin build script project. I realized actually I included by myself the `jooq-meta`  dependency in v3.16 in my build script because the plugin DSL uses directly its classes. By reverting `jooq-meta`'s version to 3.15.1 in the build script, it worked as expected.

As a result, a better fix than my last commit on all this would be to expose `jooq-meta` in the same version as in the build script directly on the plugin.

```groovy
dependencies {
    api 'org.jooq:jooq-codegen:3.15.1'
    api 'org.jooq:jooq-meta:3.15.1'
   // [...]
}
``` 

The drawback to not keep the fix in this PR is that only the version used during the compilation of the plugin can be used to configure codegen. With the fix in the PR we can override the version of jooq used to compile the plugin by yours and keep it compatible with versions pre and post 3.16. I don't know enough jOOQ for now to know if it could make a big difference between versions.

Reply:
Update: just putting the dependency in the `buildscript`  block don't break either. Maybe using `buildSrc` because that's what I do on my project...